### DISCUSSION TITLE
大表格用页面横置，landscape后的页眉页脚没有自动转置如何解决？

### DISCUSSION THREAD

**Discussion by linelee02** [sjtug/SJTUThesis#875] [提问] [answered]
大表格用页面横置，landscape后的页眉页脚没有自动转置如何解决？
另，landscape的页边距如何调整？

**Comment by LogCreative** [comment, marked as answer]
如果页眉页脚不转置，可以参见 #779 。页眉页脚转置只是方便电子版，印刷时可能会带来麻烦，慎重使用。

参考 [TeX.SX 上的回答](https://tex.stackexchange.com/a/374533)，并结合默认的 `geometry` 配置，

https://github.com/sjtug/SJTUThesis/blob/700093d999dc6fe212aa280d04aee6c120470f3b/texmf/tex/latex/sjtuthesis/sjtuthesis-graduate.ltx#L26-L34

可以给出一个临时的解决方法：

导言区添加
```latex
% \usepackage{etoolbox}
\makeatletter
\def\ifGm@preamble#1{\@firstofone}
\appto\restoregeometry{%
  \pdfpagewidth=\paperwidth
  \pdfpageheight=\paperheight
  \headwidth=\textwidth}
\apptocmd\newgeometry{%
  \pdfpagewidth=\paperwidth
  \pdfpageheight=\paperheight
  \headwidth=\textwidth}{}{}
\makeatother
```

正文需要横置的地方：
```latex
\newgeometry{,
  top        = 3.5cm,
  bottom     = 4.0cm,
  left       = 3.3cm,
  right      = 2.8cm,
  headheight = 1.0cm,
  headsep    = 0.5cm,
  landscape
}

% 横置内容

\restoregeometry
```

<img width="686" alt="截屏2022-12-23 11 52 19" src="https://user-images.githubusercontent.com/61653082/209267601-d8b3f36e-d153-42d7-9b17-de774e47a1ff.png">
